x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/support/VariablesTests.java

```diff
@@ -44,9 +48,13 @@ public void testCreateCtxModel() throws Exception {
         assertThat(model, notNullValue());
         assertThat(model.size(), is(1));
 
+        JodaCompatibleZonedDateTime jodaJavaExecutionTime =
+            new JodaCompatibleZonedDateTime(Instant.ofEpochMilli(executionTime.getMillis()), ZoneOffset.UTC);
         assertThat(ObjectPath.eval("ctx", model), instanceOf(Map.class));
         assertThat(ObjectPath.eval("ctx.id", model), is(wid.value()));
-        assertThat(ObjectPath.eval("ctx.execution_time", model), is(executionTime));
+        // NOTE: we use toString() here because two ZonedDateTime are *not* equal, we need to check with isEqual
+        // for date/time equality, but not hamcrest matcher exists
```

Comment:
but *the* hamcrest matcher doesn't exist

Reply:
Judging from a quick peek into `JodaCompatibleZonedDateTime`, it seems that its `equals` methods is the culprit not the `ZonedDateTime`? I think this snippet would fail

```
JodaCompatibleZonedDateTime d = new JodaCompatibleZonedDateTime(Instant.now(), ZoneOffset.UTC);
assertThat(d.equals(d), is(true));
```

not sure this is intentional?

Reply:
It is semi-intentional, in that it just compares the underlying ZonedDateTime instances. To compare times, apparently isEqual is the thing to use...